cloud/src/autogluon/cloud/predictor/multimodal_cloud_predictor.py

```diff
@@ -0,0 +1,143 @@
+            you must specify the column name corresponding to image paths.
+            Images have to live in the same directory specified by the column.
+        accept: str, default = application/x-parquet
+            Type of accept output content.
+            Valid options are application/x-parquet, text/csv, application/json
+
+        Returns
+        -------
+        Pandas.DataFrame
+        Predict results in DataFrame
+        """
+        assert self.endpoint, 'Please call `deploy()` to deploy an endpoint first.'
+        assert accept in VALID_ACCEPT, f'Invalid accept type. Options are {VALID_ACCEPT}.'
+
+        import numpy as np
+
+        if isinstance(test_data, str):
+            image = cv2.imread(test_data)
+            if image is None:  # not an image
+                test_data = load_pd.load(test_data)
+            else:
+                test_data = np.array([read_image_bytes_and_encode(test_data)], dtype='object')
+                content_type = 'application/x-npy'
+        if isinstance(test_data, list):
+            images = []
+            test_data = np.array([read_image_bytes_and_encode(image) for image in images], dtype='object')
+            content_type = 'application/x-npy'
+        if isinstance(test_data, pd.DataFrame):
+            if test_data_image_column is not None:
+                test_data = convert_image_path_to_encoded_bytes_in_dataframe(test_data, test_data_image_column)
+            content_type = 'application/x-parquet'
+
+        # Providing content type here because sagemaker serializer doesn't support change content type dynamically.
+        # Pass to `endpoint.predict()` call as `initial_args` instead
+        return self._predict_real_time(test_data=test_data, accept=accept, ContentType=content_type)
+
+    def predict(
+        self,
+        test_data,
+        test_data_image_column=None,
+        image_modality_only=False,
+        **kwargs,
+    ):
+        """
+        test_data: str
+            The test data to be inferenced.
+            Can be a pandas.DataFrame, a local path or a s3 path to a csv file.
+            When predicting multimodality with image modality:
+                You need to specify `test_data_image_column`, and make sure the image column contains relative path to the image.
+            When predicting with only images:
+                User has to specify `image_modality_only` for this mode.
+                Can be a local path or a s3 path to a directory containing the images.
+        test_data_image_column: Optional(str)
+            If test_data involves image modality, you must specify the column name corresponding to image paths.
+            Images have to live in the same directory specified by the column.
+        image_modality_only: Bool
+            Whether the test_data only contains image modality.
+            If so, test data can be a local path or a s3 path to a directory containing the images.
```

Comment:
Could we infer `image_modality_only` instead of letting users specify it? If we detect a local folder or a s3 bucket of images, we may say it is image only. If it's a s3 bucket, we may need to download the files anyway.